src/tribol/physics/CommonPlane.cpp

```diff
@@ -157,302 +156,301 @@ real ComputeGapRatePressure( ContactPlaneManager& cpMgr,
+        {
+          case KINEMATIC_CONSTANT: 
+          {
+              // pre-multiply each spring stiffness by each mesh's penalty scale
+              auto stiffness1 = pen_scale1 * mesh1.getElementData().m_penalty_stiffness;
+              auto stiffness2 = pen_scale2 * mesh2.getElementData().m_penalty_stiffness;
+              // compute the equivalent contact penalty spring stiffness per area
+              penalty_stiff_per_area  = ComputePenaltyStiffnessPerArea( stiffness1, stiffness2 );
+              break;
+          }
+          case KINEMATIC_ELEMENT:
+          {
+              // add tiny_length to element thickness to avoid division by zero
+              auto t1 = mesh1.getElementData().m_thickness[ index1 ] + pen_enfrc_options.tiny_length;
+              auto t2 = mesh2.getElementData().m_thickness[ index2 ] + pen_enfrc_options.tiny_length;
+
+              if (t1 < 0. || t2 < 0.)
+              {
+                neg_thickness[0] = true;
+                err[0] = 1;
+              }
+
+              // compute each element spring stiffness. Pre-multiply the material modulus 
+              // (i.e. material stiffness) by each mesh's penalty scale
+              auto stiffness1 = pen_scale1 * mesh1.getElementData().m_mat_mod[ index1 ] / t1;
+              auto stiffness2 = pen_scale2 * mesh2.getElementData().m_mat_mod[ index2 ] / t2;
+              // compute the equivalent contact penalty spring stiffness per area
+              penalty_stiff_per_area = ComputePenaltyStiffnessPerArea( stiffness1, stiffness2 );
+              break;
+          }
+          default:
+              // no-op, quiet compiler
+              break;
+        } // end switch on kinematic penalty calculation option
+
+        ////////////////////////////////////////////////////
+        // Compute contact pressure(s) on current overlap // 
+        ////////////////////////////////////////////////////
+
+        // compute total pressure based on constraint type
+        RealT totalPressure = 0.;
+        plane.m_pressure = gap * penalty_stiff_per_area; // kinematic contribution
+        switch(pen_enfrc_options.constraint_type)
+        {
+          case KINEMATIC_AND_RATE:
+          {
+              // kinematic contribution
+              totalPressure += plane.m_pressure;
+              // add gap-rate contribution
+              totalPressure += 
+                ComputeGapRatePressure( plane, mesh1, mesh2, penalty_stiff_per_area,
+                                        pen_enfrc_options.rate_calculation );
+              break;
+          }
+          case KINEMATIC:
+              // kinematic gap pressure contribution  only
+              totalPressure += plane.m_pressure;
+              break;
+          default:
+              // no-op
+              break;
+        } // end switch on registered penalty enforcement option
+
+        // debug prints. Comment out for now, but keep for future common plane 
```

Comment:
Keep these commented out per the original code, but perhaps guard with a TRIBOL_USE_HOST in case a developer does uncomment for testing/debugging?

Reply:
I'd say it's better to leave it out.  Otherwise, a developer might uncomment this, run it on device, and get no output which could leave to confusion.  It's likely better for compilation to fail in this case.

Reply:
Even more confusing, if the code is built on host and device, a user might run on host and expect output, but it won't be there since it's guarded.